### PR TITLE
Wired up publishing to new SearchAndCompare endpoint.

### PR DESCRIPTION
**Pull request by MartinBentley**
### Context
The SearchAndCompare publish course endpoint now takes a list of course to publish
### Changes proposed in this pull request
The PublishService now has two methods for publish course and publish courses

### CODE REVIEW

File: src/ManageCourses.Api/Services/EnrichmentService.cs [hunk 1]
```diff
@@ -178,24 +178,30 @@ public bool PublishInstitutionEnrichment(string instCode, string email)
         /// <param name="instCode">institution code for the enrichment</param>
         /// <param name="ucasCourseCode">can be null</param>
         /// <param name="email">email of the user</param>
+        /// <param name="publishableOnly">if there is a draft enrichment then get the ucas contact details</param>
         /// <returns>An enrichment object with the enrichment data (if found). Nul if not found</returns>
-        public UcasCourseEnrichmentGetModel GetCourseEnrichment(string instCode, string ucasCourseCode, string email)
+        public UcasCourseEnrichmentGetModel GetCourseEnrichment(string instCode, string ucasCourseCode, string email, bool publishableOnly)
```
Comment (defong) [inline src/ManageCourses.Api/Services/EnrichmentService.cs]: Can we make this a private and have two public for the false and true

File: src/ManageCourses.Api/Services/Publish/SearchAndCompareService.cs [hunk 2]
```diff
@@ -29,57 +30,77 @@ public SearchAndCompareService(ISearchAndCompareApi api, ICourseMapper courseMap
...
-                 return false;
-             }
-             var returnBool = false;
-             try
-             {
-                 var ucasInstData = _dataService.GetUcasInstitutionForUser(email, instCode);
-                 var orgEnrichmentData = _enrichmentService.GetInstitutionEnrichment(instCode, email, true);
-                 var ucasCourseData = _dataService.GetCourse(email, instCode, courseCode);
-                 var courseEnrichmentData = _enrichmentService.GetCourseEnrichment(instCode, courseCode, email);
+        public async Task<bool> SaveCourse(string instCode, string courseCode, string email)
```
Comment (defong) [inline src/ManageCourses.Api/Services/Publish/SearchAndCompareService.cs]: Whats the point?

File: src/ManageCourses.Api/Services/EnrichmentService.cs (hunk 1 above)
Comment (MartinBentley) [reply src/ManageCourses.Api/Services/EnrichmentService.cs]: This creates a further refactor issue so we'll card this up to be done at a later date

File: src/ManageCourses.Api/Services/Publish/SearchAndCompareService.cs (hunk 2 above)
Comment (MartinBentley) [reply src/ManageCourses.Api/Services/Publish/SearchAndCompareService.cs]: Refactored the code in the service

File: src/ManageCourses.Api/Services/Publish/ISearchAndCompareService.cs [hunk 3]
```diff
@@ -7,6 +7,7 @@ namespace GovUk.Education.ManageCourses.Api.Services.Publish
 {
     public interface ISearchAndCompareService
     {
-        Task<bool> SaveSingleCourseToSearchAndCompare(string instCode, string courseCode, string email);
+        Task<bool> SaveCourse(string instCode, string courseCode, string email);
+        Task<bool> SaveCourses(string instCode, string email);
```
Comment (defong) [inline src/ManageCourses.Api/Services/Publish/ISearchAndCompareService.cs]: No Test

Comment (MartinBentley) [reply src/ManageCourses.Api/Services/Publish/ISearchAndCompareService.cs]: Tests added

File: src/ManageCourses.Api/ManageCourses.Api.csproj [hunk 4]
```diff
@@ -8,7 +8,7 @@
     <DocumentationFile>bin\Debug\$(TargetFramework)\$(MSBuildProjectName).xml</DocumentationFile>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="DFE.SearchAndCompare.Domain" Version="0.8.0-beta" />
+    <PackageReference Include="DFE.SearchAndCompare.Domain" Version="0.9.0-beta" />
```
Comment (defong) [inline src/ManageCourses.Api/ManageCourses.Api.csproj]: need to up to latest please

Comment (MartinBentley) [reply src/ManageCourses.Api/ManageCourses.Api.csproj]: Done

**Pull request merged**